modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/util/DLValidatorImpl.java

```diff
@@ -69,6 +69,13 @@ public String fixName(String name) {
 		return _replaceDLWebDAVSubstitutionChar(name);
 	}
 
+	@Override
+	public long getMaxAllowableSize(
+		long externalMaxSize, long groupId, String mimeType) {
+
+		return _min(externalMaxSize, getMaxAllowableSize(groupId, mimeType));
+	}
```

Comment:
Alphabetical sorting is only mandatory for private methods. Can you rename `externalMaxSize` to `limit` and take it as a last argument? This will make the method signature more similar to the one of the already existing method, so it looks like a natural overload.

Reply:
Also, can you make the 3 argument method do all the work. Then, the two argument method would be much simpler, something like:
```
return getMaxAllowedSize(groupId, mimeType, 0);
```

Reply:
Yes milord :)

Reply:
It is "your highness" for you, petty worm

Reply:
Sorry your Highness :P